accession-commons-mongodb/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/mongodb/document/AccessionedDocument.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.persistence.mongodb.document;
+
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.domain.Persistable;
+import org.springframework.data.mongodb.core.index.Indexed;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.IAccessionedObject;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
```

Comment:
don't you need a `@Document` here? oh, I see this is abstract

Reply:
Yes, the idea is the same as in the jpa implementation, you extend and add the @Document to that other class, that way we don't need to fight the configuration to define the names of the collections in the end application.

Reply:
Could this and `AccessionedEntity` be documented as something like the following?

> Base class for accessioned objects to be serialized as [relational entities | MongoDB documents]. The derived classes must be annotated as [Entity | Document].

Picking the most appropriate option for each database.